src/pages/Top.js

```diff
@@ -9,10 +9,23 @@ const Top = () => {
   const [permission, setPermission] = useState('');
   const [accountData, setAccountData] = useState({});
 
+  const usePrevious = (value) => {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  }
+
+  const prevAmount = usePrevious({ error });
```

Comment:
Let's change `prevAmount` to `prevState` to follow the naming convention of the `componentDidUpdate` arguments.